### PR TITLE
Fixing paths and variables for RetroDECK

### PR DESCRIPTION
**Pull request by XargonWan**

### CODE REVIEW

File: PortMaster/retrodeck/PortMaster.txt [hunk 1]
```diff
@@ -23,7 +24,7 @@ echo "Starting PortMaster."
 chmod -R +x .
 
 #check RetrodDECK installed and read location of roms folder
-if [[ -e "${HOME}/.var/app/net.retrodeck.retrodeck/config/retrodeck/retrodeck.cfg" ]]; then
+if [[ -e "/app/bin/retrodeck.sh" ]]; then
```
Comment (XargonWan) [inline PortMaster/retrodeck/PortMaster.txt:27]: Actually when /app/libexec/global.sh is sourced, roms folder is just $roms_folder, PortMaster games are in $roms_folder/portmaster for us, and not inside ports as it's a different system.
PortMaster will be its own "system" in ES-DE.

File: PortMaster/pylibs/harbourmaster/config.py [hunk 2]
```diff
@@ -94,11 +94,11 @@
     HM_DEFAULT_PORTS_DIR   = Path("/storage/roms/ports")
     HM_DEFAULT_SCRIPTS_DIR = Path("/storage/roms/ports")
 
-## Check if retrodeck.cfg exists. Chose this file/location as platform independent from were retrodeck is installed.
-elif (Path.home() / ".var/app/net.retrodeck.retrodeck/config/retrodeck/retrodeck.cfg").is_file():
-    retrodeck_roms_path = retrodeck_roms_path = (Path.home() / ".var/app/net.retrodeck.retrodeck/config/retrodeck/retrodeck.cfg").read_text()
+## Check if retrodeck.sh exists. Chose this file/location as platform independent from were retrodeck is installed.
+elif ("/var/app/bin/retrodeck.sh").is_file():
+    retrodeck_roms_path = retrodeck_roms_path = ("/var/config/retrodeck/retrodeck.cfg").read_text() # is it possible to read from the bash variable $roms_dir and elaborarte that as done in the .sh files?
```
Comment (XargonWan) [inline PortMaster/pylibs/harbourmaster/config.py:99]: is it possible to read from the bash variable $roms_dir and elaborate that as done in the .sh files?

File: PortMaster/retrodeck/control.txt [hunk 3]
```diff
@@ -1,16 +1,12 @@
...
-# Referenced by RetroDeck when the app is running. Might be needed at future point.
-#source /app/libexec/global.sh
-
-
-if [ -d "${HOME}/.var/app/net.retrodeck.retrodeck/config/retrodeck/" ]; then
-  controlfolder=$(grep 'roms_folder' "${HOME}/.var/app/net.retrodeck.retrodeck/config/retrodeck/retrodeck.cfg" | awk -F= '{print $2}')"/ports/PortMaster"
+if [ -d "/app/bin/retrodeck.sh" ]; then
+  source /app/libexec/global.sh # loading the RetroDECK framework that even give access to variables such as roms_folder
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+  directory="$roms_folder/portmaster"
```
Comment (kloptops) [inline PortMaster/retrodeck/control.txt:6]: This means ports will live in: `$roms_folder/portmaster/ports/`, is that acceptable?

File: PortMaster/pylibs/harbourmaster/config.py [hunk 4]
```diff
@@ -94,11 +94,11 @@
     HM_DEFAULT_PORTS_DIR   = Path("/storage/roms/ports")
     HM_DEFAULT_SCRIPTS_DIR = Path("/storage/roms/ports")
 
-## Check if retrodeck.cfg exists. Chose this file/location as platform independent from were retrodeck is installed.
-elif (Path.home() / ".var/app/net.retrodeck.retrodeck/config/retrodeck/retrodeck.cfg").is_file():
-    retrodeck_roms_path = retrodeck_roms_path = (Path.home() / ".var/app/net.retrodeck.retrodeck/config/retrodeck/retrodeck.cfg").read_text()
+## Check if retrodeck.sh exists. Chose this file/location as platform independent from were retrodeck is installed.
+elif ("/var/app/bin/retrodeck.sh").is_file():
```
Comment (kloptops) [inline PortMaster/pylibs/harbourmaster/config.py:98]: Small typo, needs to be a Path object:
```suggestion
elif Path("/var/app/bin/retrodeck.sh").is_file():
```

File: PortMaster/pylibs/harbourmaster/config.py (hunk 2 above)
Comment (kloptops) [inline PortMaster/pylibs/harbourmaster/config.py]: Same here, also double assignment?
```suggestion
    retrodeck_roms_path = Path("/var/config/retrodeck/retrodeck.cfg").read_text() # is it possible to read from the bash variable $roms_dir and elaborarte that as done in the .sh files?
```

**Review by kloptops — CHANGES_REQUESTED** [review]
Has a few small issues, also the `$roms_path/portmaster/` may or may not cause issues. :)

File: PortMaster/pylibs/harbourmaster/config.py [hunk 5]
```diff
@@ -95,13 +96,23 @@
...
 ## Check if retrodeck.sh exists. Chose this file/location as platform independent from were retrodeck is installed.
-elif ("/var/app/bin/retrodeck.sh").is_file():
-    retrodeck_roms_path = retrodeck_roms_path = ("/var/config/retrodeck/retrodeck.cfg").read_text() # is it possible to read from the bash variable $roms_dir and elaborarte that as done in the .sh files?
-    if retrodeck_roms_path != '':
-        retrodeck_roms_path = retrodeck_roms_path.join(re.findall(r'roms_folder=(.*)', retrodeck_roms_path)) + "/portmaster"
-        HM_DEFAULT_TOOLS_DIR   = Path(retrodeck_roms_path)
-        HM_DEFAULT_PORTS_DIR   = Path(retrodeck_roms_path)
-        HM_DEFAULT_SCRIPTS_DIR = Path(retrodeck_roms_path)
+elif ('roms_dir' in os.environ) or ("/var/config/retrodeck/retrodeck.cfg").is_file():
+    if 'roms_dir' in os.environ:
```
Comment (kloptops) [inline PortMaster/pylibs/harbourmaster/config.py:100]: Add default here:
```suggestion
    retrodeck_roms_path    = str(Path().home() / 'retrodeck/roms/ports')
    if 'roms_dir' in os.environ:
```

File: PortMaster/pylibs/harbourmaster/config.py [hunk 6]
```diff
@@ -30,6 +30,7 @@
 HM_DEFAULT_PORTS_DIR   = Path("/roms/ports")
 HM_DEFAULT_SCRIPTS_DIR = Path("/roms/ports")
 HM_DEFAULT_TOOLS_DIR   = Path("/roms/ports")
+retrodeck_roms_path    = str(Path().home() / 'retrodeck/roms/ports')
```
Comment (kloptops) [inline PortMaster/pylibs/harbourmaster/config.py:33]: Doesn't need to be top level, keep it in the relevant area. :)
```suggestion
```

**Review by kloptops — CHANGES_REQUESTED** [review]

File: PortMaster/retrodeck/control.txt (hunk 3 above)
Comment (XargonWan) [reply PortMaster/retrodeck/control.txt:6]: Probably not optimal actually but might work for the moment. However I might find a way to tweak it later.
My idea is to have this structure:

$roms
- portmaster
 - PortMaster.sh/PortMaster.sh
 - game1.sh/game1.sh
 - game99.sh/game99.sh
 
In this way both the games and PortMaster will be recognized ad files a folders, and cloud be run with 1 button press, but I have understand how much this is possible.

**Pull request merged**